fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.api.rdf;
+
+import static org.fcrepo.kernel.api.FedoraTypes.FCR_ACL;
+import static org.fcrepo.kernel.api.FedoraTypes.FCR_VERSIONS;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import com.google.common.base.Converter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import javax.ws.rs.core.UriBuilder;
+import org.glassfish.jersey.uri.UriTemplate;
+import org.slf4j.Logger;
+
+/**
+ * Convert between HTTP URIs (LDP paths) and internal Fedora ID using a
+ * JAX-RS UriBuilder to mediate the URI translation.
+ *
+ * @author whikloj
+ * @since 2019-09-26
+ */
+public class HttpIdentifierConverter extends Converter<String, String> {
+
+    private static final Logger LOGGER = getLogger(HttpIdentifierConverter.class);
+
+    private final UriBuilder uriBuilder;
+
+    protected Converter<String, String> forward = identity();
+    protected Converter<String, String> reverse = identity();
+
+    private final UriTemplate uriTemplate;
+
+    private static final String FEDORA_ID_PREFIX = "info:fedora/";
+
+    /**
+     * Split uri on delim and return the first part.
+     */
+    private static BiFunction<String, String, String> removeDelim =
+        (uri, delim) -> {
+            final String newUri;
+            if (uri.contains(delim)) {
+                newUri = uri.split(delim)[0];
+            } else {
+                newUri = uri;
+            }
+            return newUri;
+        };
+
+    /**
+     * Create a new identifier converter within the given session with the given URI template
+     * @param uriBuilder the uri builder
+     */
+    public HttpIdentifierConverter(final UriBuilder uriBuilder) {
+
+        this.uriBuilder = uriBuilder;
+        this.uriTemplate = new UriTemplate(uriBuilder.toTemplate());
+    }
+
+    private UriBuilder uriBuilder() {
+        return UriBuilder.fromUri(uriBuilder.toTemplate());
+    }
+
+    @Override
+    protected String doForward(final String httpUri) {
+        LOGGER.debug(String.format("Translating http URI %s to Fedora ID", httpUri));
+
+        final String path = getPath(httpUri);
+        if (path != null) {
+
+            // Take the URL and remove any hash uris, or fcr: endpoints, except fcr:metadata.
+            final String fedoraId = Stream.of(path).map(p -> removeDelim.apply(p, "#"))
```

Comment:
Looking at this again, i'm wondering if using the streaming api for method chaining on a single item is a good thing for readability. Maybe it'd make sense to define something like 
`private final static ID_SUFFIXES = { "#", "/" + FCR_ACL, "/" + FCR_VERSIONS };`
and then make a `truncateSuffixes` method that loops through and returns that path with them?